shell/browser/native_browser_view_mac.mm

```diff
@@ -133,13 +133,14 @@ - (void)mouseDragged:(NSEvent*)event {
   [self.window setFrameOrigin:newOrigin];
 }
 
-// For debugging purposes only.
-- (void)drawRect:(NSRect)aRect {
-  if (getenv("ELECTRON_DEBUG_DRAG_REGIONS")) {
```

Comment:
Instead of calling `getenv` in every `drawRect`, could caching the result in the constructor fix the flicker?

```objective-c
- (id)initWithFlag:(BOOL)enableDebug {
  enableDebug_ = enableDebug;
  ...
}

- (void)drawRect:(NSRect)aRect {
  if (enableDebug_) {
    ...
  }
}
```

If overriding the method itself is causing problems, another option is to dynamically override the `drawRect` method with `class_replaceMethod`, you can find some examples in `native_window_mac.mm`.

Reply:
Ill try this!